addons/adyen_platforms/models/adyen_account.py

```diff
@@ -311,412 +393,245 @@ def _adyen_rpc(self, operation, adyen_data={}):
+                transactions, next_page = account._fetch_transactions(page)
+                page += 1
+
+                for transaction in transactions:
+                    create_date = parse(transaction.get('creationDate')).astimezone(UTC).replace(tzinfo=None)
+                    if not max_create_date:
+                        max_create_date = create_date
+                    if create_date <= account.last_sync_date:
+                        next_page = False
+                        break
+
+                    status = transaction.get('transactionStatus')
+                    if status in ['Payout', 'PayoutReversed']:
+                        reference = transaction.get('pspReference') or transaction.get('disputePspReference')
+                        tx_sudo = account.transaction_payout_ids.sudo().filtered(lambda t: t.reference == reference)
+                        if not tx_sudo:
+                            self.env['adyen.transaction.payout'].sudo()._create_missing_payout(account.id, transaction)
+                        else:
+                            tx_sudo.status = status
+                    else:
+                        if status in ['Chargeback', 'ChargebackReceived']:
+                            transaction['pspReference'] = transaction.get('disputePspReference')
+                        tx_sudo, _reference, _capture_reference = account.transaction_ids.sudo()._get_tx_from_notification(transaction)
+                        if not tx_sudo:
+                            tx_sudo = self.env['adyen.transaction'].sudo()._create_missing_tx(account.id, transaction)
+                        tx_sudo._update_status(status, create_date)
+                        updated_transactions |= tx_sudo
+                account.last_sync_date = max_create_date
+        updated_transactions.sudo()._post_transaction_sync()
 
     def _fetch_transactions(self, page=1):
-        response = self.adyen_account_id._adyen_rpc('get_transactions', {
-            'accountHolderCode': self.adyen_account_id.account_holder_code,
+        self.ensure_one()
+        response = self._adyen_rpc('v1/get_transactions', {
+            'accountHolderCode': self.account_holder_code,
             'transactionListsPerAccount': [{
-                'accountCode': self.code,
+                'accountCode': self.account_code,
                 'page': page,
             }]
         })
         transaction_list = response['accountTransactionLists'][0]
         return transaction_list['transactions'], transaction_list['hasNextPage']
+
+
+class AdyenAccountBalance(models.Model):
+    _name = 'adyen.account.balance'
+    _description = 'Adyen Account Balance'
+
+    adyen_account_id = fields.Many2one('adyen.account', required=True, ondelete='cascade')
+    currency_id = fields.Many2one('res.currency')
+    balance = fields.Float(default=0.0)
+    on_hold = fields.Float(default=0.0)
+    pending = fields.Float(default=0.0)
+
+    @api.model
+    def get_account_balance(self):
```

Comment:
Doesn't this method needs to be protected? Can anyone call it? Verification of groups?

Reply:
It was crashing on the read of `company.adyen_account_id` for users without access rights, but I've added an explicit check now :-)